### PR TITLE
use assumptions to ignore slack integration tests without env params

### PR DESCRIPTION
**Pull request by dsvoronin**
enabler for local build.sh run

### CODE REVIEW

File: subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt [hunk 1]
```diff
@@ -87,4 +97,9 @@ internal class SlackConditionalSenderIntegrationTest {
             emoji = ":crazy-robot:"
         )
     ).onFailure { it.printStackTrace() }
+
+    private fun String?.assumeNotNullOrBlank(): String {
+        Assumptions.assumeFalse { isNullOrBlank() }
```
Comment (eugene-krivobokov) [inline subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:102]: It looks like a half measure.
What are we fixing? Local checks?

1. Such optimizations give a false sense of safety in local runs.
2. We can't do a fail-fast check in CI. 
Imagine a situation when we lose these ENVs in CI. Tests are green but we expected a failure.

Let's choose a more explicit way.

**Review by eugene-krivobokov — CHANGES_REQUESTED** [review]

File: subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:102]: any suggestions?

Comment (dsvoronin) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:102]: tests will be ignored, not green.
We can forbid ignored tests at all in CI

Comment (eugene-krivobokov) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:102]: > We can forbid ignored tests at all in CI

In this case, we need to distinguish explicitly `@Disabled` tests from implicitly skipped by assumptions.

I propose another solution, to pass an intention through parameters, e.g.:

- Default value in CI properties: `SLACK_INTEGRATION_TESTS` == enabled -> requires other parameters (url, token, ...).
- Default value in the project: `SLACK_INTEGRATION_TESTS` == disabled -> skip tests by assumptions
- Else (unset or unknown value) -> fail fast, require to pass the intention. 

We use a similar pattern in gradle plugins. 
It protects us from accidentally lost parameters. 

Also, I like to separate concerns when feature enabled and not configured properly.
If I don't have SLACK_TEST_CHANNEL it shouldn't mean that I don't want to run tests. It's harder to understand and triage.

File: subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt [hunk 2]
```diff
@@ -6,16 +6,26 @@ import com.avito.slack.model.SlackSendMessageRequest
...
-    private val testChannel = SlackChannel(requireNotNull(System.getenv("SLACK_TEST_CHANNEL")))
-    private val testToken = requireNotNull(System.getenv("SLACK_TEST_TOKEN"))
-    private val slackClient: SlackClient =
-        SlackClient.Impl(testToken, requireNotNull(System.getenv("SLACK_TEST_WORKSPACE")))
+    private val testChannel by lazy {
+        System.getenv("SLACK_TEST_CHANNEL").assumeNotNullOrBlank().let { SlackChannel(it) }
+    }
+
+    private val testToken by lazy {
+        System.getenv("SLACK_TEST_TOKEN").assumeNotNullOrBlank()
```
Comment (sergioboi) [inline subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:20]: i think we must mark it Depreceted and try to pass parameters via constructor

Comment (eugene-krivobokov) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:20]: Do you know another way to pass a value to a unit-test?

Comment (sergioboi) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:20]: Constructor. We can abstract env logic from this class becuase it is dependency details leak

Comment (eugene-krivobokov) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:20]: Constructor of what exactly? 
I don't see a difference. We need to get value for tests.

Comment (sergioboi) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:20]: Sorry i don' understand. Why do we need env for test?
I want hide token impl details.

**Review by sergioboi — CHANGES_REQUESTED** [review]

File: subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:20]: it's an integration test. we need slack hook token to execute it

Comment (sergioboi) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:20]: Can we get slack data not from environment?

Comment (dsvoronin) [reply subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt:20]: not much of a choice: either env or gradle param

**Review by sergioboi — APPROVED** [review]

**Pull request closed without merging**